onnxruntime/core/providers/rocm/rocm_provider_factory.cc

```diff
@@ -46,3 +47,18 @@ ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProvider_ROCM, _In_ OrtSessi
 
   return nullptr;
 }
+
+ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider_ROCM,
+                    _In_ OrtSessionOptions* options, _In_ const OrtROCMProviderOptions* rocm_options) {
+  ROCMExecutionProviderInfo info{};
+  info.device_id = gsl::narrow<OrtDevice::DeviceId>(rocm_options->device_id);
+  info.hip_mem_limit = rocm_options->hip_mem_limit;
+  info.arena_extend_strategy = static_cast<onnxruntime::ArenaExtendStrategy>(rocm_options->arena_extend_strategy);
+  // info.miopen_conv_algo_search = rocm_options->miopen_conv_algo_search;
```

Comment:
Is this line applicable to MIOpen? If not, it is better to delete it.

Reply:
I changed it to a boolean type, which is what the miopen API uses.

Reply:
Probably you can't change to a boolean type. I remember the build will fail with boolean type for C API.

Reply:
Should I change it to an int?

Reply:
you can check CI result and then decide.

Reply:
Looks like the CI build was not done, do we need to do something to trigger it?

Reply:
[443/978] Building C object CMakeFiles/onnxruntime4j_jni.dir/onnxruntime_src/java/src/main/native/OrtJniUtil.c.o
FAILED: CMakeFiles/onnxruntime4j_jni.dir/onnxruntime_src/java/src/main/native/OrtJniUtil.c.o 
/usr/bin/cc -DEIGEN_MPL2_ONLY -DENABLE_ORT_FORMAT_LOAD -DNSYNC_ATOMIC_CPP11 -DORT_RUN_EXTERNAL_ONNX_TESTS -DPLATFORM_POSIX -DUSE_EIGEN_FOR_BLAS -Donnxruntime4j_jni_EXPORTS -I/onnxruntime_src/include/onnxruntime -I/onnxruntime_src/include/onnxruntime/core/session -I/usr/lib/jvm/java/include -I/usr/lib/jvm/java/include/linux -I/onnxruntime_src/include -I/onnxruntime_src/java/build/headers -fopenmp -Wall -Wextra -ffunction-sections -fdata-sections -Werror -O3 -DNDEBUG -fPIC -std=gnu99 -MD -MT CMakeFiles/onnxruntime4j_jni.dir/onnxruntime_src/java/src/main/native/OrtJniUtil.c.o -MF CMakeFiles/onnxruntime4j_jni.dir/onnxruntime_src/java/src/main/native/OrtJniUtil.c.o.d -o CMakeFiles/onnxruntime4j_jni.dir/onnxruntime_src/java/src/main/native/OrtJniUtil.c.o -c /onnxruntime_src/java/src/main/native/OrtJniUtil.c
In file included from /onnxruntime_src/java/src/main/native/OrtJniUtil.h:6:0,
                 from /onnxruntime_src/java/src/main/native/OrtJniUtil.c:7:
/onnxruntime_src/include/onnxruntime/core/session/onnxruntime_c_api.h:278:3: error: unknown type name 'bool'
   bool miopen_conv_exhaustive_search;               // miopen conv algo exhaustive search option
   ^

Reply:
Thanks. Changed to int type.